src/commands/appSettings/SqlDatabaseConnectionCustomPromptStep.ts

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import { ConnectionType } from '../../constants';
+import { getInvalidLengthMessage } from '../../constants-nls';
+import { localize } from '../../localize';
+import { validateUtils } from '../../utils/validateUtils';
+import { ISqlDatabaseConnectionWizardContext } from './ISqlDatabaseConnectionWizardContext';
+
+export class SqlDatabaseConnectionCustomPromptStep<T extends ISqlDatabaseConnectionWizardContext> extends AzureWizardPromptStep<T> {
+    public async prompt(context: T): Promise<void> {
+        context.nonAzureSqlConnection = (await context.ui.showInputBox({
+            prompt: localize('customSqlConnectionPrompt', 'Enter a custom SQL connection string.'),
```

Comment:
Yeah fair.  It was originally intended to cover the `non-Azure` scenarios, but the user could definitely use it to cover an Azure scenario as well since it just replaces the string directly.